### ISSUE TITLE
Add Mongo expression interpreter

### ISSUE THREAD

**Issue by mark-kar** [rafafrdz/criteria4s#14]
It will be nice to add support for mongoDb expressions.
At least Postgres + real NoSql solution can be really beautiful example of using this algebra

**Comment by rafafrdz** [comment]
sure! do you want to try? 😃

**Comment by mark-kar** [comment]
Yeah, I've already have some code, but at first I think it'll be better to finalize an algebra
